src/requirements/VisitLink/VisitLinkRequirement.tsx

```diff
@@ -0,0 +1,110 @@
+import { HStack, Icon, Stack, Text } from "@chakra-ui/react"
+import Requirement, {
+  RequirementProps,
+} from "components/[guild]/Requirements/components/Requirement"
+import { useRequirementContext } from "components/[guild]/Requirements/components/RequirementContext"
+import {
+  RequirementImage,
+  RequirementImageCircle,
+} from "components/[guild]/Requirements/components/RequirementImage"
+import ResetRequirementButton from "components/[guild]/Requirements/components/ResetRequirementButton"
+import ViewOriginalPopover from "components/[guild]/Requirements/components/ViewOriginalPopover"
+import useAccess from "components/[guild]/hooks/useAccess"
+import useUser from "components/[guild]/hooks/useUser"
+import Link from "components/common/Link"
+import useShowErrorToast from "hooks/useShowErrorToast"
+import { SignedValdation, useSubmitWithSign } from "hooks/useSubmit"
+import { Link as LinkIcon } from "phosphor-react"
+import { ComponentProps } from "react"
+import fetcher from "utils/fetcher"
+
+export const VISIT_LINK_REGEX = new RegExp(/^(.*)(\[)(.+?)(\])(.*)$/)
+
+const visitLink = (signedValidation: SignedValdation) =>
+  fetcher("/v2/util/gate-callbacks?requirementType=LINK_VISIT", {
+    ...signedValidation,
+    method: "POST",
+  })
+
+const VisitLinkRequirement = ({ ...props }: RequirementProps) => {
+  const { id: requirementId, data } = useRequirementContext()
+  const { id: userId } = useUser()
+
+  const { data: accesses, mutate: mutateAccess } = useAccess()
+  const hasAccess = accesses
+    ?.flatMap((role) => role.requirements)
+    .find((req) => req.requirementId === requirementId)?.access
+
+  const showErrorToast = useShowErrorToast()
+
+  const { onSubmit } = useSubmitWithSign(visitLink, {
+    onSuccess: () => mutateAccess(),
+    onError: () => showErrorToast("Something went wrong"),
+  })
+
+  const [, first, , link, , second] = !!data.customName
+    ? VISIT_LINK_REGEX.exec(data.customName) ?? []
+    : []
+
+  const linkProps: ComponentProps<typeof Link> = {
+    href: data.id,
+    isExternal: true,
+    colorScheme: "blue",
+    onClick: () => {
+      if (!userId || hasAccess) return
+      onSubmit({
+        requirementId,
+        id: data.id,
+        userId,
+      })
+    },
+  }
+
+  const Original = () => (
+    <>
+      {"Visit link: "}
+      <Link {...linkProps}>{data.id}</Link>
+    </>
+  )
+
+  return (
+    <Requirement
+      image={<Icon as={LinkIcon} boxSize={6} />}
+      {...props}
+      showViewOriginal={false}
+      footer={
```

Comment:
So I basically overwrote the default "View original" popover here, because this requirement type won't ever have a real default value (see `Requirement.tsx`), but let me explain the other solution which I had in my mind:

We could add an optional `childrenWrapperComponent` prop to `requirements.ts` and make a `VisitLinkChildrenParser` component which would to the following stuff:
- check if there's a string child
- if there isn't, just return children
- if there is, it could try to parse it (see lines 45-47 & 97-105 in this file) and display the parsed value

Not sure which would be the cleaner solution, that's why I sticked with the first one... but lmk what's your opinion on this!